liblangutil/ErrorReporter.h

```diff
@@ -87,6 +90,23 @@ class ErrorReporter
 
 	void typeError(SourceLocation const& _location, std::string const& _description);
 
+	template <typename... Strings>
+	void typeErrorConcatenateDescriptions(SourceLocation const& _location, Strings const&... _descriptions)
+	{
+		std::initializer_list<std::string> const descs = {_descriptions...};
```

Comment:
I'm wondering whether this performs a copy. Would `std::initializer_list<std::string const&>` work, too?

Reply:
I tried using `const&` but I get a bunch of errors when compiling that:

```
[ 15%] Building CXX object liblangutil/CMakeFiles/langutil.dir/ParserBase.cpp.o
In file included from /usr/include/c++/8.2.1/bits/stl_vector.h:63,
                 from /usr/include/c++/8.2.1/vector:64,
                 from /home/marenz/ethereum/solidity-wt0/libdevcore/vector_ref.h:6,
                 from /home/marenz/ethereum/solidity-wt0/libdevcore/Common.h:40,
                 from /home/marenz/ethereum/solidity-wt0/liblangutil/Token.h:45,
                 from /home/marenz/ethereum/solidity-wt0/liblangutil/ParserBase.h:25,
                 from /home/marenz/ethereum/solidity-wt0/liblangutil/ParserBase.cpp:23:
/usr/include/c++/8.2.1/initializer_list: In Instanziierung von »class std::initializer_list<const std::__cxx11::basic_string<char>&>«:
/home/marenz/ethereum/solidity-wt0/liblangutil/ErrorReporter.h:97:3:   von hier erfordert
/usr/include/c++/8.2.1/initializer_list:54:26: Fehler: Zeiger auf Referenztyp »const std::__cxx11::basic_string<char>&« wird geformt
       typedef const _E*  iterator;
                          ^~~~~~~~
/usr/include/c++/8.2.1/initializer_list:55:26: Fehler: Zeiger auf Referenztyp »const std::__cxx11::basic_string<char>&« wird geformt
       typedef const _E*  const_iterator;
                          ^~~~~~~~~~~~~~
make[2]: *** [liblangutil/CMakeFiles/langutil.dir/build.make:102: liblangutil/CMakeFiles/langutil.dir/ParserBase.cpp.o] Fehler 1
make[2]: *** Es wird auf noch nicht beendete Prozesse gewartet....
In file included from /usr/include/c++/8.2.1/bits/stl_vector.h:63,
                 from /usr/include/c++/8.2.1/vector:64,
                 from /home/marenz/ethereum/solidity-wt0/libdevcore/vector_ref.h:6,
                 from /home/marenz/ethereum/solidity-wt0/libdevcore/Common.h:40,
                 from /home/marenz/ethereum/solidity-wt0/libdevcore/CommonData.h:26,
                 from /home/marenz/ethereum/solidity-wt0/liblangutil/ErrorReporter.h:25,
                 from /home/marenz/ethereum/solidity-wt0/liblangutil/ErrorReporter.cpp:23:
/usr/include/c++/8.2.1/initializer_list: In Instanziierung von »class std::initializer_list<const std::__cxx11::basic_string<char>&>«:
/home/marenz/ethereum/solidity-wt0/liblangutil/ErrorReporter.h:97:3:   von hier erfordert
/usr/include/c++/8.2.1/initializer_list:54:26: Fehler: Zeiger auf Referenztyp »const std::__cxx11::basic_string<char>&« wird geformt
       typedef const _E*  iterator;
                          ^~~~~~~~
/usr/include/c++/8.2.1/initializer_list:55:26: Fehler: Zeiger auf Referenztyp »const std::__cxx11::basic_string<char>&« wird geformt
       typedef const _E*  const_iterator;
                          ^~~~~~~~~~~~~~
make[2]: *** [liblangutil/CMakeFiles/langutil.dir/build.make:76: liblangutil/CMakeFiles/langutil.dir/ErrorReporter.cpp.o] Fehler 1
make[2]: Verzeichnis „/home/marenz/ethereum/solidity-wt0/build“ wird verlassen
make[1]: *** [CMakeFiles/Makefile2:222: liblangutil/CMakeFiles/langutil.dir/all] Fehler 2
make[1]: Verzeichnis „/home/marenz/ethereum/solidity-wt0/build“ wird verlassen
make: *** [Makefile:130: all] Fehler 2
make: Verzeichnis „/home/marenz/ethereum/solidity-wt0/build“ wird verlassen
```